src/cppcache/include/geode/SharedBase.hpp

```diff
@@ -56,11 +57,12 @@ class CPPCACHE_EXPORT SharedBase {
 
  protected:
   inline SharedBase(bool noInit) {}
+  inline SharedBase(const SharedBase&) {}
 
   virtual ~SharedBase() {}
 
  private:
-  mutable volatile int32_t m_refCount;
+  std::atomic<int32_t> m_refCount;
```

Comment:
Is the mutable unnecessary? Or does std::atomic have const operators that can be used to mutate the reference count?